src/components/Button/Button.js

```diff
@@ -0,0 +1,97 @@
+import styled from 'styled-components';
+
+const Button = ({
+  type = 'button',
+  shape = 'solid',
+  color = 'neutral',
+  size = 'medium',
+  full = false,
+  content = 'button',
+  onClick,
+  disabled = false,
+}) => {
+  return (
+    <DefaultButton
+      className="btn"
+      type={type}
+      shape={shape}
+      color={color}
+      size={size}
+      full={full}
+      aria-label={content}
+      onClick={onClick}
+      disabled={disabled}
+    >
+      {content}
+    </DefaultButton>
+  );
+};
+
+/**
+ * Button props list
+ * @property {string} type: button, submit, reset           - 버튼 타입을 정의합니다.
+ * @property {string} shape: solid, outline, round          - 버튼 형태를 정의합니다.
+ * @property {string} color: primary, secondary, neutral    - 버튼 색상을 정의합니다.
+ * @property {string} size: small, medium, large            - 버튼 크기를 정의합니다.
+ * @property {boolean} full                                 - 버튼 가로 너비를 뷰포트에 가득 채울 것인지 정의합니다.
+ * @property {string} content                               - 버튼 내부 텍스트와 웹 접근성 처리에 사용합니다.
+ * @property {function} onClick                             - 버튼 클릭 시 실행할 함수를 위해 미리 정의합니다.
+ * @property {boolean} disabled                             - 버튼의 비활성화 상태를 정의합니다.
+ */
+
+// 버튼 기본 구조를 스타일링합니다. 모든 버튼에서 사용할 수 있게 css로 처리하고, ${ButtonBasicStructure}처럼 mixin으로 꺼내쓰게 합니다.
+const DefaultButton = styled.button`
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  border: 1px transparent solid;
+  opacity: 0.9;
+  cursor: pointer;
+  width: ${props => (props.full ? '100%' : 'auto')};
+  min-width: ${props =>
+    (props.size === 'small' && '150px') ||
+    (props.size === 'medium' && '175px') ||
+    '200px'};
+  padding: ${props =>
+    (props.size === 'small' && '11px 10px') ||
+    (props.size === 'medium' && '12px 10px') ||
+    '13px 10px'};
+  font-size: ${props =>
+    (props.size === 'small' && '14px') ||
+    (props.size === 'medium' && '16px') ||
+    '18px'};
```

Comment:
고치시는 김에 ㅎㅎ.. 이런 식으로 해보는 건 어떨까요? 맵핑하여 사용하면 가독성도 더 좋고 관리하기도 좋을 것 같습니다.

```suggestion
const SIZE_STYLES = {
  small: {
    minWidth: '150px',
    padding: '11px 10px',
    fontSize: '14px',
  },
  medium: {
    minWidth: '175px',
    padding: '12px 10px',
    fontSize: '16px',
  },
  large: {
    minWidth: '200px',
    padding: '13px 10px',
    fontSize: '18px',
  },
};

...

  min-width: ${({size}) => SIZE_STYLES[size]?.minWidth || '200px'};
  padding: ${({size}) => SIZE_STYLES[size]?.padding || '13px 10px'};
  font-size: ${({size}) => SIZE_STYLES[size]?.fontSize || '18px'};
```

Reply:
혹은 css helper를 활용해서 매핑해볼 수도 있겠네요!

Reply:
완료했습니다.